Sources/FoundationNetworking/URLSession/URLSessionDelegate.swift

```diff
@@ -244,15 +246,19 @@ public protocol URLSessionDataDelegate : URLSessionTaskDelegate {
 
 extension URLSessionDataDelegate {
     
-    public func urlSession(_ session: URLSession, dataTask: URLSessionDataTask, didReceive response: URLResponse, completionHandler: @escaping (URLSession.ResponseDisposition) -> Void) { }
+    public func urlSession(_ session: URLSession, dataTask: URLSessionDataTask, didReceive response: URLResponse, completionHandler: @escaping (URLSession.ResponseDisposition) -> Void) {
+        completionHandler(.allow)
```

Comment:
There was a recent change that added the API for `URLSessionTask.delegate`. A delegate set on task has priority over the delegate set on the session, but if the task-specific delegate doesn't implement the function, then we see if the delegate set on the session does. We can do this here and in the other function below, using the pattern

```
if self === task.delegate, let sessionDelegate = session.delegate as? URLSessionDataDelegate, self !== sessionDelegate {
    sessionDelegate.urlSession(session, dataTask: dataTask, didReceive: response, completionHandler: completionHandler)
} else {
    completionHandler(.allow)
}
```